runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -330,6 +330,11 @@ class MM_ObjectAllocationAPI
 		/* Calculate the size of the object */
 		uintptr_t const headerSize = J9VMTHREAD_OBJECT_HEADER_SIZE(currentThread);
 		uintptr_t dataSize = clazz->totalInstanceSize;
+		bool ensureHashed = J9_ARE_ANY_BITS_SET(clazz->classFlags, J9ClassEnsureHashed);
+		if (ensureHashed) {
+			/* Add extra uintptr_t for hash */
+			dataSize += sizeof(uintptr_t);
```

Comment:
This is incorrect - we are not assigning the hash here, just tagging it so the GC will add the hash slot if the object moves.